rackcity/views/asset_views.py

```diff
@@ -164,49 +170,66 @@ def asset_add(request):  # need to make network and power connections here
-        try:
-            validate_asset_location(rack_id, rack_position, height)
-        except LocationException as error:
-            failure_message += str(error)
-            return JsonResponse(
-                {"failure_message": failure_message},
-                status=HTTPStatus.BAD_REQUEST,
-            )
-    if failure_message == "":
-        try:
-            asset = serializer.save()
-        except Exception as error:
-            failure_message += str(error)
-        else:
-            try:
-                save_mac_addresses(
-                    asset_data=data,
-                    asset_id=asset.id
-                )
-            except MacAddressException as error:
-                failure_message += "Some mac addresses couldn't be saved. " + \
-                    str(error)
-            try:
-                save_power_connections(
-                    asset_data=data,
-                    asset_id=asset.id
-                )
-            except PowerConnectionException as error:
-                failure_message += "Some power connections couldn't be saved. " + \
-                    str(error)
-    if failure_message:
         return JsonResponse(
             {"failure_message": failure_message},
             status=HTTPStatus.BAD_REQUEST,
         )
+    rack_id = serializer.validated_data['rack'].id
+    rack_position = serializer.validated_data['rack_position']
+    height = serializer.validated_data['model'].height
+    try:
+        validate_asset_location(rack_id, rack_position, height)
+    except LocationException as error:
+        failure_message += str(error)
+        return JsonResponse(
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    try:
+        asset = serializer.save()
+    except Exception as error:
+        failure_message += str(error)
+        return JsonResponse(
+            {"failure_message": failure_message},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    warning_message = ""
+    try:
+        save_mac_addresses(
+            asset_data=data,
+            asset_id=asset.id
+        )
+    except MacAddressException as error:
+        warning_message += "Some mac addresses couldn't be saved. " + \
+            str(error)
+    try:
+        save_power_connections(
+            asset_data=data,
+            asset_id=asset.id
+        )
+    except PowerConnectionException as error:
+        warning_message += "Some power connections couldn't be saved. " + \
+            str(error)
+    try:
+        save_network_connections(
+            asset_data=data,
+            asset_id=asset.id
+        )
+    except NetworkConnectionException as error:
+        warning_message += "Some network connections couldn't be saved. " + \
+            str(error)
+    if warning_message:
+        return JsonResponse(
+            {"warning_message": warning_message},
+            status=HTTPStatus.OK,
+        )
```

Comment:
🚨🚨🚨 flagging to @clairecfu !!! I'm open to discussing other ways to do this but essentially this is what we talked about that there's a new case we have to handle in forms where the asset was created or modified successfully (form should not stay open) but we failed to create all mac addresses, power connections, and/or network connections. The backend returns a very descriptive message in these cases but we can't just throw it away because it was a status=200. I looked into whether there's a warning like HTTP status (personally I think it would be called `status=HTTPStatus.OK_BUT`) but it doesn't exist.  We could agree on a unique status for these cases though